Source/cryptalgo/Hash.h

```diff
@@ -115,17 +131,33 @@ namespace Crypto {
         static const EnumHashType Type = HASH_SHA1;
         static const uint8_t Length = 20;
 
+        static uint32_t ContextLength()
```

Comment:
If you make the interface (constructor with the uint8_t) typesafe (take a SHA1::Context ) this method is not needed.

Reply:
done